recipes/unix.rb

```diff
@@ -0,0 +1,190 @@
+#
+# Cookbook Name:: android-sdk
+# Recipe:: unix
+#
+# Copyright 2011-2013, Travis CI Development Team <contact@travis-ci.org>
+# Authors: Gilles Cornu
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+setup_root       = node['android-sdk']['setup_root'].to_s.empty? ? node['ark']['prefix_home'] : node['android-sdk']['setup_root']
+android_home     = File.join(setup_root, node['android-sdk']['name'])
+android_bin      = File.join(android_home, 'tools', 'android')
+
+#
+# Install required libraries
+#
+if node['platform'] == 'ubuntu'
+  package 'libgl1-mesa-dev'
+
+  #
+  # Install required 32-bit libraries on 64-bit platforms
+  #
+  if node['kernel']['machine'] != 'i686'
+    # http://askubuntu.com/questions/147400/problems-with-eclipse-and-android-sdk
+    if Chef::VersionConstraint.new('>= 13.04').include?(node['platform_version'])
+      package 'lib32stdc++6'
+    elsif Chef::VersionConstraint.new('>= 11.10').include?(node['platform_version'])
+      package 'libstdc++6:i386'
+    end
+
+    package 'lib32z1'
+  end
+end
+
+#
+# Download and setup android-sdk tarball package
+#
+if linux?
+  sdk_url = node['android-sdk']['download_url']['linux']
+  sdk_checksum = node['android-sdk']['checksum']['linux']
+elsif mac_os_x?
+  sdk_url = node['android-sdk']['download_url']['mac_os_x']
+  sdk_checksum = node['android-sdk']['checksum']['mac_os_x']
+else
+  Chef::Log.error('Platform not supported!')
+end
+
+ark node['android-sdk']['name'] do
+  url sdk_url
+  path node['android-sdk']['setup_root']
+  checksum sdk_checksum
+  version node['android-sdk']['version']
+  prefix_root node['android-sdk']['setup_root']
+  prefix_home node['android-sdk']['setup_root']
+  owner node['android-sdk']['owner']
+  group node['android-sdk']['group'] unless mac_os_x?
```

Comment:
Don't think this should be forced to be `unless mac_os_x` . Takes control away from downstream cookbooks.